### PR TITLE
Fixes #19130 - dzdo support

### PR DESCRIPTION
**Pull request by witlessbird**
This PR depends on https://github.com/theforeman/foreman_remote_execution/pull/342

### CODE REVIEW

**Comment by iNecas** [comment]
Rubocop is not happy

**Comment by witlessbird** [comment]
* rebased
* corrected rubocop warning

**Comment by witlessbird** [comment]
* rebased

**Comment by witlessbird** [comment]
* rebased

**Comment by iNecas** [comment]
[test foreman_remote_execution]

**Comment by iNecas** [comment]
While testing I got:

```
/home/inecas/Projects/ws/foreman-rex/foreman_remote_execution/app/lib/actions/remote_execution/run_host_job.rb:124:in `update_host_status'
/home/inecas/Projects/ws/foreman-rex/foreman_remote_execution/app/lib/actions/remote_execution/run_host_job.rb:57:in `finalize'
/home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/action.rb:560:in `block (2 levels) in execute_finalize'
/home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/middleware/stack.rb:26:in `pass'
/home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/middleware.rb:18:in `pass'
/home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/middleware.rb:39:in `finalize'
/home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/middleware/stack.rb:22:in `call'
/home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/middleware/stack.rb:26:in `pass'
```

Seems like the Dzdo implementation is missing some methods. Given the Sudo and Dzdo are so similar, doesn't it makes sense for the Dzdo to inherit from Sudo and just override the changes that differ?

**Comment by iNecas** [comment]
While testing this, I've also hit the issue with the `loop` vs `process` method https://github.com/theforeman/foreman_remote_execution/pull/351 - happened to me when dzdo was not present on the system.

**Comment by witlessbird** [comment]
* ```DzdoUserMethod``` inherits from ```SudoUserMethod``` now.

File: lib/foreman_remote_execution_core/script_runner.rb [hunk 1]
```diff
@@ -44,6 +44,14 @@ def reset
     end
   end
 
+  class DzdoUserMethod < SudoUserMethod
+    LOGIN_PROMPT = /password/i
```
Comment (iNecas) [inline lib/foreman_remote_execution_core/script_runner.rb]: This actually doesn't get used. Converting this from constant to method should help: I guess the penalty of creating a regex object every time vs. having that in constant is acceptable.

**Comment by witlessbird** [comment]
* fixed issues pointed out by @iNecas

**Comment by iNecas** [comment]
[test foreman_remote_execution]

**Comment by iNecas** [comment]
Tested and works well. Waiting for tests to finish.

**Comment by iNecas** [comment]
Thanks @witlessbird

**Pull request merged**